### ISSUE TITLE
not able to simulate the test case. its always capture the response

### ISSUE THREAD

**Issue by antika08** [SpectoLabs/hoverfly-java#200]
package Junit;

import static org.junit.Assert.assertThat;
import static org.junit.Assert.fail;
import java.util.Set;
import static io.specto.hoverfly.junit.core.HoverflyConfig.localConfigs;
import static io.specto.hoverfly.junit.core.HoverflyMode.*;
import static io.specto.hoverfly.junit.core.SimulationSource.file;

import io.restassured.RestAssured;
import io.restassured.specification.RequestSpecification;
import io.restassured.response.Response;
import io.specto.hoverfly.junit.core.HoverflyConfig;
import io.specto.hoverfly.junit.core.model.*;
import io.specto.hoverfly.junit.dsl.StubServiceBuilder;
import io.specto.hoverfly.junit.dsl.*;
import io.specto.hoverfly.junit.rule.HoverflyRule;
import io.specto.hoverfly.junit.core.config.*;
import io.specto.hoverfly.junit.verification.*;
import io.specto.hoverfly.junit.dsl.matchers.*;
import io.specto.hoverfly.junit.api.*;
import org.junit.Test;
import static org.junit.Assert.assertEquals;

import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.ClassRule;
import org.junit.Rule;
import org.junit.rules.*;

public class test {

//	@Autowired
//	LoginController lcontroller;

	@ClassRule
	public static HoverflyRule hoverflyRule = HoverflyRule
			.inCaptureOrSimulationMode("response.json", HoverflyConfig.localConfigs().proxyLocalHost())
		    .printSimulationData();
	
	
	@Test
public void test2(){
		
		RestAssured.baseURI = "http://time.jsontest.com/";
		//RestAssured.baseURI = "http://restapi.demoqa.com/utilities/weather/city/Hyderabad";
		
        Response res;
		res=RestAssured.given()

		.relaxedHTTPSValidation()

		.when()

		.get()
		
		.then()

		.contentType("application/json")

		.and()

		.extract().response();

		String responseBodyGet=res.asString();    
		System.out.println("Response body: "+responseBodyGet);
	}

	
	@Test
	public void test4(){
			
			RestAssured.baseURI = "https://jsonplaceholder.typicode.com/todos/1";
			//RestAssured.baseURI = "http://restapi.demoqa.com/utilities/weather/city/Hyderabad";
			
	        Response res;
			res=RestAssured.given()

			.relaxedHTTPSValidation()

			.when()

			.get()
			
			.then()

			.contentType("application/json")

			.and()

			.extract().response();

			String responseBodyGet=res.asString();    
			System.out.println("Response body: "+responseBodyGet);
		}

	

i 
}

**Comment by tommysitu** [comment]
It's likely that the `RestAssured` requests were not proxied through Hoverfly. 

Http clients must honour the JAVA system proxy settings in order for HoverflyRule to work. However, this is probably not the case for `RestAssured` according to this issue here: https://github.com/rest-assured/rest-assured/issues/734

You can try setting RestAssured to use proxy: 
```
RestAssured.proxy = host("127.0.0.1").withPort(hoverflyRule.getProxyPort());
```

**Comment by antika08** [comment]
still facing same issue. :(

**Comment by tommysitu** [comment]
Does it create the `response.json` file? The `inCaptureOrSimulationMode` checks for the existence of the file to determine what mode to use.

**Comment by antika08** [comment]
its create the response.json file.
but inCaptureOrSimulationMode does not check properly the existence of the file.
response.json file is exist it should not goes into simulation mode.
its always goes into capture mode

**Comment by antika08** [comment]
please help me regarding this

**Comment by tommysitu** [comment]
sure, can you tell me the path of the `response.json` that the capture mode creates? It should be created under `hoverfly` folder in your classpath or test resources folder. 

You can also try debugging into `hoverfly-java`, and set a break point here https://github.com/SpectoLabs/hoverfly-java/blob/master/src/main/java/io/specto/hoverfly/junit/rule/HoverflyRule.java#L119

**Comment by antika08** [comment]
Capture Path:\src\test\resources\hoverfly

**Comment by tommysitu** [comment]
It looks ok. Have you tried debugging? Hoverfly does the following check before setting to simulate mode: 
` if (path.isPresent() && Files.isRegularFile(path.get())) `

I wonder which bit is evaluated to false for you. Without access to your environment, I can't tell, and I cannot recreate the issue from my end.

**Comment by antika08** [comment]
Hoverfly is capture the response on \src\test\resources\hoverfly path
but while simulating the response its will check on path target\test-classes\hoverfly for maven project

i dont know why is working like this
please check this.

**Comment by antika08** [comment]
[junitrestweb.zip](https://github.com/SpectoLabs/hoverfly-java/files/28637Pleas35/junitrestweb.zip)


this is code which i am using....please check this once and let me know

**Comment by tommysitu** [comment]
Will take a look sometimes this week.

**Comment by antika08** [comment]
this is very urgent for me. please take a look at the code and help me

**Comment by tommysitu** [comment]
The problem is that when using maven test, it doesn't add `test/resources` to the classpath, therefore Hoverfly cannot find the captured response: https://stackoverflow.com/questions/3571149/test-resources-of-dependencies-not-in-classpath

The solution is creating a new overloaded `inCaptureOrSimulateMode` method that accepts an absolute

**Comment by tommysitu** [comment]
Try 0.11.3-SNAPSHOT version, it contains a fix

**Comment by antika08** [comment]
i had tried to use 0.11.3-SNAPSHOT version of hoverfly-java in maven pom file, but i am getting error "Missing artifact io.specto:hoverfly-java:jar:0.11.3-SNAPSHOT"


<dependency>
			<groupId>io.specto</groupId>
			<artifactId>hoverfly-java</artifactId>
			<version>0.11.3-SNAPSHOT</version>
		</dependency>

**Comment by tommysitu** [comment]
You probably missing the snapshot repo in your maven config: https://hoverfly-java.readthedocs.io/en/latest/pages/misc/misc.html#using-snapshot-version

**Comment by tommysitu** [comment]
Does it fix your problem @antika08 ?

**Comment by tommysitu** [comment]
I believed that this is now fixed in 0.11.3-SNAPSHOT
